internal/controller/application_controller.go

```diff
@@ -97,22 +99,31 @@ func (r *ApplicationPullServiceAccountCreator) Reconcile(ctx context.Context, re
 		return ctrl.Result{}, err
 	}
 
-	// service account doesn't exist we will have to create it
-	err = r.createApplicationServiceAccount(ctx, applicationSaName, application, pullSecretNames)
+	aggregatedSecretName := getAggregatedPullSecretName(application.Name)
+	aggregatedSecret, err := r.createOrUpdateAggregatedPullSecret(ctx, aggregatedSecretName, application, pullSecretNames)
```

Comment:
also try to get applicationSecret before that function AND if it exists, just return nil

image-controller will already be keeping it updated once it exists